lib/components/util/styledIcon.tsx

```diff
@@ -0,0 +1,62 @@
+import { StyledIconBase } from '@styled-icons/styled-icon'
+import styled, { keyframes } from 'styled-components'
+
+interface Props {
+  flipHorizontal?: boolean
+  noMargin?: boolean
+  rotate90?: boolean
+  size?: string
+  spaceRight?: boolean
+  spin?: boolean
+  static?: boolean
+}
+
+const getFontSize = (size?: string) => {
+  switch (size) {
+    case '1.5x':
+      return '1.5em'
+    case '2x':
+      return '2em'
+    case '3x':
+      return '3em'
+    case '4x':
+      return '4em'
+    case '5x':
+      return '5em'
+    default:
+      return 'inherit'
+  }
+}
+
+const rotateAnimation = keyframes`
+  0% {
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+`
+
+const StyledIconWrapper = styled.span<Props>`
+  top: 0.125em;
+  position: ${(props) => (props.static ? 'static' : 'relative')};
+  display: inline-flex;
```

Comment:
Could we do `justify-self` as well? It's still rendering a bit off center (vertically) for me.
<img width="290" alt="Screen Shot 2022-08-17 at 10 05 47 AM" src="https://user-images.githubusercontent.com/86619099/185154845-d42d1b77-92eb-49ae-a89a-1c57a228ac4b.png">
s